src/parse.rs

```diff
@@ -73,16 +74,16 @@ const fn is_float_char(c: u8) -> bool {
     ENCODINGS[c as usize] & FLOAT_CHAR != 0
 }
 
-pub const fn is_ident_first_char(c: u8) -> bool {
-    ENCODINGS[c as usize] & IDENT_FIRST_CHAR != 0
+pub fn is_ident_first_char(c: char) -> bool {
+    c == '_' || is_xid_start(c)
 }
 
-pub const fn is_ident_other_char(c: u8) -> bool {
-    ENCODINGS[c as usize] & IDENT_OTHER_CHAR != 0
-}
-
-pub const fn is_ident_raw_char(c: u8) -> bool {
-    ENCODINGS[c as usize] & IDENT_RAW_CHAR != 0
+pub fn is_ident_raw_char(c: char) -> bool {
+    if c.is_ascii() {
+        ENCODINGS[c as usize] & IDENT_RAW_CHAR != 0
```

Comment:
Which of the initial ascii byte classes for the encoding table are still necessary?

Reply:
we can probably shrink that a bit. I was also thinking about doing a performance comparison with std functions like [`is_digit`](https://doc.rust-lang.org/std/primitive.char.html#method.is_digit)

> I am wondering if we could not entirely stop relying on bytes but replace all "next byte" methods with "next character" methods. Would this be possible? What are your thoughts @ModProg?